Fatras/Geant4/src/Geant4Decay.cpp

```diff
@@ -0,0 +1,77 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsFatras/Geant4/Geant4Decay.hpp"
+
+#include "Acts/Definitions/Common.hpp"
+#include "ActsFatras/EventData/ProcessType.hpp"
+#include "ActsFatras/Geant4/DummyDetectorConstruction.hpp"
+
+#include "G4DecayProducts.hh"
+#include "G4DecayTable.hh"
+
+ActsFatras::Geant4Decay::Geant4Decay()
+    : m_g4RunManager(makeDummyRunManager()) {}
+
+std::vector<ActsFatras::Particle> ActsFatras::Geant4Decay::decayParticle(
+    const ActsFatras::Particle& parent) const {
+  std::vector<Particle> children;
+
+  // Find the particle type that will decay
+  G4ParticleDefinition* pDef =
+      m_pdgToG4Conv.getParticleDefinition(parent.pdg());
+  if (!pDef) {
+    return children;
+  }
+  // Get the particles decay table
+  G4DecayTable* dt = pDef->GetDecayTable();
+  if (!dt) {
+    return children;
+  }
+  // Select a decay channel
+  G4VDecayChannel* channel = dt->SelectADecayChannel();
+  if (!channel) {
+    return children;
+  }
+  // Get the decay products from the selected channel
+  G4DecayProducts* products = channel->DecayIt();
```

Comment:
Who owns the product pointers? Is Geant4 handling or do they need to be deleted manually?

Reply:
I cannot be 100% sure as it is very hard to track down such details in the G4 code but it looks to me that this is part of the physics stored in the run manager. So I assume that the run manager itself takes care of it.

Reply:
Ok, lets see if it does.